application.js

```diff
@@ -46,31 +48,85 @@ var sessionOptions = {
   }
 };
 
+/**
+ * Gets mongodb connection information from $fh.db
+ * Use Data Browser > Upgrade Database to get the connection information available
+ */
+function mongoInit(cb) {
+  $fh.db({
+    act: 'connectionString'
+  }, function(err, connectionString) {
+    if (err) {
+      return cb(err);
+    }
+    if (!connectionString) {
+      console.log('$fh.db connectionString not found, assuming database on localhost');
+    } else {
+      console.log('$fh.db connectionString found');
+      sessionOptions.config.url = connectionString;
+    }
+    cb();
+  });
+}
+
+/**
+ * Initialize parameters for redis client from default env vars supplied
+ * by the platform
+ */
+function redisInit(cb) {
+  sessionOptions.config.host = process.env.FH_REDIS_HOST || '127.0.0.1',
+  sessionOptions.config.port = process.env.FH_REDIS_PORT || '6379',
+  cb();
+}
+
 // find out mongodb connection string from $fh.db
-$fh.db({
-  act: 'connectionString'
-}, function(err, connectionString) {
-  if (err || !connectionString) {
-    console.log('$fh.db connectionString not found, assuming database on localhost');
-  } else {
-    sessionOptions.url = connectionString;
+function run(cb) {
+  var initFn = function(callback) {
+    callback();
+  };
+  switch (sessionOptions.store) {
+  case 'redis':
+    initFn = redisInit;
+    break;
+  case 'mongo':
```

Comment:
Probably good to have a default case here if neither redis or mongo is specified.